.github/scripts/branch_pr_issue_closer.py

```diff
@@ -135,176 +98,165 @@ def _main_prog():
+        merged_branch = merged_pull.base.ref
 
-    #If PR was to default branch, then exit script (as github will handle it automatically):
-    if merged_branch == default_branch:
-        endmsg = "Pull request ws merged into default repo branch. Thus issue is closed automatically"
-        end_script(endmsg)
+        #If PR was to default branch, then exit script (as github will handle it automatically):
+        if merged_branch == default_branch:
+            endmsg = f"Pull request #{pr_num} was merged into default repo branch. "
+            endmsg += "Thus issue is closed automatically"
+            end_script(endmsg)
 
-    #++++++++++++++++++++++++++++++++++++++
-    #Create integer list of all open issues:
-    #++++++++++++++++++++++++++++++++++++++
+        #++++++++++++++++++++++++++++++++++++++
+        #Create integer list of all open issues:
+        #++++++++++++++++++++++++++++++++++++++
 
-    #Extract list of open issues from repo:
-    open_repo_issues = cam_repo.get_issues(state='open')
+        #Extract list of open issues from repo:
+        open_repo_issues = cam_repo.get_issues(state='open')
 
-    #Collect all open repo issues:
-    open_issues = [issue.number for issue in open_repo_issues]
+        #Collect all open repo issues:
+        open_issues = [issue.number for issue in open_repo_issues]
 
-    #+++++++++++++++++++++++++++++++++++++++++++++
-    #Create integer list of all open pull requests
-    #+++++++++++++++++++++++++++++++++++++++++++++
+        #+++++++++++++++++++++++++++++++++++++++++++++
+        #Create integer list of all open pull requests
+        #+++++++++++++++++++++++++++++++++++++++++++++
 
-    #Extract list of open PRs from repo:
-    open_repo_pulls = cam_repo.get_pulls(state='open')
+        #Extract list of open PRs from repo:
+        open_repo_pulls = cam_repo.get_pulls(state='open')
 
-    #Collect all open pull requests:
-    open_pulls = [pr.number for pr in open_repo_pulls]
+        #Collect all open pull requests:
+        open_pulls = [pr.number for pr in open_repo_pulls]
 
-    #+++++++++++++++++++++++++++++++++++++++++++++++++
-    #Check if one of the keywords exists in PR message
-    #+++++++++++++++++++++++++++++++++++++++++++++++++
+        #+++++++++++++++++++++++++++++++++++++++++++++++++
+        #Check if one of the keywords exists in PR message
+        #+++++++++++++++++++++++++++++++++++++++++++++++++
 
-    #Keywords are:
-    #close, closes, closed
-    #fix, fixes, fixed
-    #resolve, resolves, resolved
+        #Keywords are:
+        #close, closes, closed
+        #fix, fixes, fixed
+        #resolve, resolves, resolved
 
-    #Create regex pattern to find keywords:
-    keyword_pattern = re.compile(r'(^|\s)close(\s|s\s|d\s)|(^|\s)fix(\s|es\s|ed\s)|(^|\s)resolve(\s|s\s|d\s)')
+        #Create regex pattern to find keywords:
+        keyword_pattern = re.compile(r'(^|\s)close(\s|s\s|d\s)|(^|\s)fix(\s|es\s|ed\s)|(^|\s)resolve(\s|s\s|d\s)')
```

Comment:
```suggestion
        _CLOSE_KEY      = r'close(s|d)?'
        _FIX_KEY        = r'fix(e(s|d))?'
        _RESOLVE_KEY    = r'resolve(s|d)?'
        _KEYWORDS       = rf'{_CLOSE_KEY}|{_FIX_KEY}|{_RESOLVE_KEY}'
        keyword_pattern = re.compile(r'(^|\s)({_KEYWORDS})\s')
```
I know this was already here but it might be worth while to break up the keywords to make it easier to track the whitespace checks as well.

Reply:
```python
import re

_CLOSE_KEY   = r'close[sd]?'
_FIX_KEY     = r'fix(e[sd])?'
_RESOLVE_KEY = r'resolve[sd]?'
_KEYWORDS = rf'{_CLOSE_KEY}|{_FIX_KEY}|{_RESOLVE_KEY}'
_ID_NUMBER = r'[1-9]\d*'
keyword_pattern = rf'(?P<keyword>{_KEYWORDS})\s*#(?P<id>{_ID_NUMBER})'
to_search = 'closed #2, Fixes #4'

for m in re.finditer(keyword_pattern, to_search, re.IGNORECASE):
    issue_dict = m.groupdict()
    print(issue_dict['keyword'], issue_dict['id'])
# prints:
# closed 2
# Fixes 4
```
@nusbaume, this should make the search code a little easier and also works for multiline strings.